lib/rouge/lexers/livescript.rb

```diff
@@ -0,0 +1,308 @@
+        rule %r/\\\S[^\s,;\])}]*/, Str
+      end
+
+      state :code_escape do
+        rule %r(\\(
+          c[A-Z]|
+          x[0-9a-fA-F]{2}|
+          u[0-9a-fA-F]{4}|
+          u\{[0-9a-fA-F]{4}\}
+        ))x, Str::Escape
+      end
+
+      state :interpolated_expression do
+        rule %r/}/, Str::Interpol, :pop!
+        mixin :root
+      end
+
+      state :interpolation do
+        # with curly braces
+        rule %r/[#][{]/, Str::Interpol, :interpolated_expression
+        # without curly braces
+        rule %r/(#)(#{id})/ do |m|
+          groups Str::Interpol, if self.class.builtins.include? m[2] then Name::Builtin else Name::Variable end
+        end
+      end
+
+      state :whitespace do
+        # white space and loop labels
+        rule %r/(\s+?)(?:^([^\S\n]*)(:#{id}))?/m do
+          groups Text, Text, Name::Label
+        end
+      end
+
+      state :whitespace_single_line do
+        rule %r([^\S\n]+), Text
+      end
+
+      state :slash_starts_regex do
+        mixin :comments
+        mixin :whitespace
+        mixin :multiline_regex_begin
+
+        rule %r(
+          /(\\.|[^\[/\\\n]|\[(\\.|[^\]\\\n])*\])+/ # a regex
+          ([gimy]+\b|\B)
+        )x, Str::Regex, :pop!
+
+        rule(//) { pop! }
+      end
+
+      state :multiline_regex_begin do
+        rule %r(//) do
+          token Str::Regex
+          goto :multiline_regex
+        end
+      end
+
+      state :multiline_regex_end do
+        rule %r(//([gimy]+\b|\B)), Str::Regex, :pop!
+      end
+
+      state :multiline_regex do
+        mixin :multiline_regex_end
+        mixin :regex_comment
+        mixin :interpolation
+        mixin :code_escape
+        rule %r/\\\D/, Str::Escape
+        rule %r/\\\d+/, Name::Variable
+        rule %r/./m, Str::Regex
+      end
+
+      state :regex_comment do
+        rule %r/^#(\s+.*)?$/, Comment::Single
+        rule %r/(\s+)(#)(\s+.*)?$/ do
+          groups Text, Comment::Single, Comment::Single
+        end
+      end
+
+      state :comments do
+        rule %r(/\*.*?\*/)m, Comment::Multiline
+        rule %r/#.*$/, Comment::Single
+      end
+
+      state :switch_underscore do
+        mixin :whitespace_single_line
+        rule %r/_(?=\s*=>|\s+then\b)/, Keyword
+        rule(//) { pop! }
+      end
+
+      state :loop_control do
+        mixin :whitespace_single_line
+        rule %r/#{id}(?=[);\n])/, Name::Label
+        rule(//) { pop! }
+      end
+
+      state :id do
+        rule %r/[!?]|[.](?!=)/, Punctuation
+        rule %r/[{]/ do
+          # destructuring
+          token Punctuation
+          push do
+            rule %r/[,;]/, Punctuation
+            rule %r/#{id}/, Name::Attribute
+            rule %r/#{int}/, Num::Integer
+            mixin :whitespace
+            rule %r/[}]/, Punctuation, :pop!
+          end
+        end
+        rule %r/#{id}/, Name::Attribute
+        rule %r/#{int}/, Num::Integer
+        rule(//) { pop!; push :slash_starts_regex }
```

Comment:
You can make this a little simpler with `goto`

```suggestion
        rule(//) { goto :slash_starts_regex }
```

I'll make the change and push it.